flexbox/src/main/java/com/google/android/flexbox/FlexboxLayout.java

```diff
@@ -715,59 +751,67 @@ private int shrinkFlexItems(FlexLine flexLine, @FlexDirection int flexDirection,
             LayoutParams lp = (LayoutParams) child.getLayoutParams();
             if (flexDirection == FLEX_DIRECTION_ROW
                     || flexDirection == FLEX_DIRECTION_ROW_REVERSE) {
-                float rawCalculatedWidth = child.getMeasuredWidth() - unitShrink * lp.flexShrink;
-                if (i == flexLine.itemCount - 1) {
-                    rawCalculatedWidth += accumulatedRoundError;
-                    accumulatedRoundError = 0;
-                }
-                int newWidth = Math.round(rawCalculatedWidth);
-                if (newWidth < 0) {
-                    // This means the child doesn't have enough space to distribute the negative
-                    // free space. To adjust the flex line length down to the maxMainSize, remaining
-                    // negative free space needs to be re-distributed to other flex items
-                    // (children views). In that case, invoke this method again with the same
-                    // startIndex.
-                    needsReshrink = true;
-                    newWidth = 0;
-                } else {
-                    accumulatedRoundError += (rawCalculatedWidth - newWidth);
-                    if (accumulatedRoundError > 1.0) {
-                        newWidth += 1;
-                        accumulatedRoundError -= 1;
-                    } else if (accumulatedRoundError < -1.0) {
-                        newWidth -= 1;
-                        accumulatedRoundError += 1;
+                if (!mFrozenChildren[childIndex]) {
```

Comment:
I think this condition check can be at the line 752 before the if statement:
```java
if (flexDirection == FLEX_DIRECTION_ROW
                      || flexDirection == FLEX_DIRECTION_ROW_REVERSE) {		                      || flexDirection == FLEX_DIRECTION_ROW_REVERSE)
```

like 
```java
if (mFrozenChildren[childIndex]) {
    childIndex++;
    continue;
}
```

To avoid the duplicate if checks in the line 754 and 788 and to reduce the indent